tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTokenInterceptor.java

```diff
@@ -0,0 +1,61 @@
+package com.epam.pipeline.tesadapter.service;
+
+import com.epam.pipeline.tesadapter.entity.TesTokenHolder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.Optional;
+
+@Slf4j
+@SuppressWarnings("unused")
+public class TesTokenInterceptor implements HandlerInterceptor {
+    private static final String HTTP_AUTH_COOKIE = "HttpAuthorization";
+
+    private TesTokenHolder tesTokenHolder;
+
+    @Value("${cloud.pipeline.token}")
+    private String defaultPipelineToken;
+
+    @Autowired
+    public TesTokenInterceptor(TesTokenHolder tesTokenHolder) {
+        this.tesTokenHolder = tesTokenHolder;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        if (checkRequestForToken(request).isPresent()) {
+            tesTokenHolder.setToken(checkRequestForToken(request).get());
+            return true;
+        } else if (checkClientHostAddress(request)) {
+            tesTokenHolder.setToken(defaultPipelineToken);
+            return true;
+        }
+        response.sendError(HttpServletResponse.SC_FORBIDDEN);
```

Comment:
have you check that we really will get 401 on client in this case?

Reply:
![image](https://user-images.githubusercontent.com/48137680/65589016-0e056f80-df91-11e9-8e24-ed5333e21dbb.png)
A stubbed checkClientHostAddress() with **false** response and in this case I actually can't open swagger-ui.html tab, seems it works fine, except for the fact that I get error 403 instead of 401.

Reply:
that because you need to change it to HttpServletResponse.SC_UNAUTHORIZED

Reply:
changed